scrapy/utils/misc.py

```diff
@@ -160,6 +160,50 @@ def create_instance(objcls, settings, crawler, *args, **kwargs):
     return instance
 
 
+def build_from_crawler(objcls, crawler, /, *args, **kwargs):
+    """Construct a class instance using its ``from_crawler`` constructor, if
+    available.
+
+    The arguments ``*args`` and ``**kwargs`` are forwarded to the constructors.
+
+    Raises ``TypeError`` if the resulting instance is ``None`` (e.g. if an
+    extension has not been implemented correctly.)
+    """
+    if crawler and hasattr(objcls, "from_crawler"):
```

Comment:
```suggestion
    if hasattr(objcls, "from_crawler"):
```

I think that only made sense when settings could be specified and crawler not.